### PR TITLE
Improve matching reliability, add test cases

### PR DESCRIPTION
**Pull request by guerler**
Follow-up for #13591. Adds query filtering tests and improves matching reliability.

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: client/src/store/historyStore/historyItemsFiltering.test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { getFilters, getQueryDict, testFilters } from "./historyItemsFiltering";
+
+const filterTexts = [
+    "name='name of item' hid>10 hid<100 create-time>'2021-01-01' update-time<\"2022-01-01\" state=success extension=ext tag=first",
+    "name='name of item' hid>10 hid<100 create_time>\"2021-01-01\" update_time-lt='2022-01-01' state=success extension=ext tag=first",
```
Comment (davelopez) [inline client/src/store/historyStore/historyItemsFiltering.test.js]: I guess you wanted to try more alternative syntax cases here?
```suggestion
    "name='name of item' hid_gt=10 hid-lt=100 create_time-gt=\"2021-01-01\" update_time-lt='2022-01-01' state=success extension=ext tag=first",
```

File: client/src/store/historyStore/historyItemsFiltering.test.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                update_time: "2021-06-01",
+                state: "success",
+                extension: "ext",
+                tags: ["first", "second"],
+            };
+            expect(testFilters(filters, { ...item })).toBe(true);
+            expect(testFilters(filters, { ...item, hid: 10 })).toBe(false);
+            expect(testFilters(filters, { ...item, hid: 100 })).toBe(false);
+            expect(testFilters(filters, { ...item, hid: 99 })).toBe(true);
+            expect(testFilters(filters, { ...item, status: "error" })).toBe(true);
```
Comment (davelopez) [inline client/src/store/historyStore/historyItemsFiltering.test.js]:
```suggestion
            expect(testFilters(filters, { ...item, state: "error" })).toBe(false);
```

File: client/src/store/historyStore/historyItemsFiltering.test.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            expect(testFilters(filters, { ...item, create_time: "2021-01-01" })).toBe(false);
+            expect(testFilters(filters, { ...item, create_time: "2021-01-02" })).toBe(true);
+            expect(testFilters(filters, { ...item, update_time: "2022-01-01" })).toBe(false);
+            expect(testFilters(filters, { ...item, update_time: "2021-12-31" })).toBe(true);
+            expect(testFilters(filters, { ...item, status: "error" })).toBe(true);
```
Comment (davelopez) [inline client/src/store/historyStore/historyItemsFiltering.test.js]: This one seems duplicated
```suggestion
```

**Review by davelopez — APPROVED** [review]
Thanks for adding some tests!

**Review by dannon — APPROVED** [review]

**Pull request merged**